internal/registry/registry.go

```diff
@@ -427,3 +375,52 @@ func (r *registry) ExtractFileToFile(destination string, header *tar.Header, tar
 	return nil
 
 }
+
+func (r *registry) getTransportOptions(insecure bool, skipTLSVerify bool) []crane.Option {
+
+	options := []crane.Option{}
+
+	if insecure {
+		options = append(options, crane.Insecure)
+	}
+
+	if skipTLSVerify {
+		rt := http.DefaultTransport.(*http.Transport).Clone()
+		rt.TLSClientConfig.InsecureSkipVerify = true
+		options = append(
+			options,
+			crane.WithTransport(rt),
+		)
+	}
+
+	return options
+}
+
+func (r *registry) WriteImageByName(imageName string, image v1.Image, auth authn.Authenticator, insecure bool, skipTLSVerify bool) error {
+	options := r.getTransportOptions(insecure, skipTLSVerify)
+	options = append(
+		options,
+		crane.WithAuth(auth),
+	)
+
+	err := crane.Push(image, imageName, options...)
+	if err != nil {
+		return fmt.Errorf("failed to push signed image: %w", err)
+	}
+	return nil
+}
+
+func (r *registry) GetImageByName(imageName string, auth authn.Authenticator, insecure bool, skipTLSVerify bool) (v1.Image, error) {
+	options := r.getTransportOptions(insecure, skipTLSVerify)
+	options = append(
+		options,
+		crane.WithAuth(auth),
+	)
+
+	img, err := crane.Pull(imageName, options...)
+	if err != nil {
+		return nil, fmt.Errorf("could not get image: %w", err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("could not get image: %v", err)
```

After some discussions, it has been agreed to not wrap errors unless the caller of the function that may return an error is using `Is(err)` or `As(err)`.